### PR TITLE
refactor fetch flow

### PR DESCRIPTION
**Pull request by almogdepaz**

### CODE REVIEW

**Comment by almogdepaz** [comment]
this is a patch and should be considered as one
AtxReqFactory and TxReqFactory no check the pool and database before each request is sent.

@zalmen if we proceed with merging this i suggest we revisit this soon

File: sync/syncer.go [hunk 1]
```diff
@@ -416,7 +416,37 @@ func (s *Syncer) validateAndBuildTx(x *types.SerializableSignedTransaction) (*ty
...
+	unprocessedTxs := make(map[types.TransactionId]*types.AddressableSignedTransaction)
+
+	for out := range s.fetchWithFactory(NewNeighborhoodWorker(s, 1, TxReqFactory(txids, s))) {
+		if ntxs, ok := out.([]types.SerializableSignedTransaction); ok {
+			for _, tmp := range ntxs {
+				tx := tmp
+				ast, err := s.validateAndBuildTx(&tx)
+				if err != nil {
+					id := types.GetTransactionId(&tx)
+					s.Warning("tx %v not valid %v", hex.EncodeToString(id[:]), err)
```
Comment (zalmen) [inline sync/syncer.go:428]: use json

File: sync/syncer.go [hunk 2]
```diff
@@ -416,7 +416,37 @@ func (s *Syncer) validateAndBuildTx(x *types.SerializableSignedTransaction) (*ty
...
+					continue
```
Comment (zalmen) [inline sync/syncer.go:429]: We found out that the block contains invalid tx, we should return err and fail the block syntactic validation.

File: sync/syncer.go [hunk 3]
```diff
@@ -428,43 +458,52 @@ func (s *Syncer) syncTxs(txids []types.TransactionId) ([]*types.AddressableSigne
 			missing = append(missing, t)
 		}
 	}
-
 	//look in db
 	dbTxs, missinDB := s.GetTransactions(missing)
+	if len(dbTxs) > 0 {
+		s.Info("found tx  in db")
```
Comment (zalmen) [inline sync/syncer.go:460]: Do you need this print as info?

File: sync/syncer.go (hunk 1 above)
Comment (zalmen) [inline sync/syncer.go]: test

**Pull request closed without merging**